include/linker/devicetree_reserved.h

```diff
@@ -57,3 +57,109 @@
  * @brief Generate linker script symbols for all the reserved memory regions
  */
 #define DT_RESERVED_MEM_SYMBOLS() _RESERVED_REGION_APPLY(_RESERVED_SYMBOL_DECLARE)
+
+/**
+ * @brief Get the pointer to the reserved-memory region
+ *
+ * Example devicetree fragment:
+ *
+ *     reserved: reserved-memory {
+ *         compatible = "reserved-memory";
+ *         ...
+ *         n: node {
+ *             reg = <0x42000000 0x1000>;
+ *         };
+ *      };
+ *
+ * Example usage:
+ *
+ *     DT_RESERVED_MEM_GET_PTR(DT_NODELABEL(n)) // (uint8_t *) 0x42000000
+ *
+ * @param node_id node identifier
+ * @return pointer to the beginning of the reserved-memory region
+ */
+#define DT_RESERVED_MEM_GET_PTR(node_id) _DT_RESERVED_START(node_id)
```

Comment:
Can you also do a sed to rename these to `LINKER_DT` instead of `DT` so we don't pollute the `DT_` namespace?

Reply:
done